Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -619,55 +619,46 @@ void FullscreenManager::dispatchPendingEvents()
     // document will be detached and GC'd. We protect it here to make sure we
     // can finish the function successfully.
     Ref<Document> protectedDocument(document());
-    Deque<GCReachableRef<Node>> changeQueue;
-    m_fullscreenChangeEventTargetQueue.swap(changeQueue);
-    Deque<GCReachableRef<Node>> errorQueue;
-    m_fullscreenErrorEventTargetQueue.swap(errorQueue);
 
-    dispatchFullscreenChangeOrErrorEvent(changeQueue, EventType::Change, /* shouldNotifyMediaElement */ true);
-    dispatchFullscreenChangeOrErrorEvent(errorQueue, EventType::Error, /* shouldNotifyMediaElement */ false);
-}
+    // Steps 1-2:
+    PendingEventsList pendingEvents;
+    m_pendingEvents.swap(pendingEvents);
```

Comment:
I think this would be better like this:
auto pendingEvents = std::exchange(m_pendingEvents, { });